### ISSUE TITLE
Type stub fails to fall back to .py when .pyi is missing

### ISSUE THREAD

**Issue by 0xjc** [microsoft/pyright#3250]
Here is a minimal version of my directory setup:

- `foo/bar.py`
- `main.py` with contents:
```
from foo import bar
```
- `typings/foo/` (empty directory)

The error:

```
/tmp/testing/main.py
  /tmp/testing/main.py:1:17 - error: "bar" is unknown import symbol (reportGeneralTypeIssues)
```

If the `typings/foo/` directory is removed, the error disappears.

My motivation for wanting the `typings/foo/` directory to exist without containing a `bar.pyi` is that for certain other files I do actually want to have type stubs, e.g. `typings/foo/baz/baz.pyi`. So if originally I have no type stubs, then I add the file `typings/foo/baz/baz.pyi`, the result is that `from foo import bar` suddenly fails.

**Comment by erictraut** [comment]
The current behavior in pyright wasn't intended; it's a side effect of some other import resolution logic designed to handle namespace packages.

Your use case is reasonable, and that's how I'd expect it to work in this case as well.

I've updated the logic so it works the way you specified above.

**Comment by erictraut** [comment]
This is included in pyright 1.1.233, which I just published. It will also be included in the next release of pylance.

**Comment by 0xjc** [comment]
Thank you!
